src/util-unittest.h

```diff
@@ -106,6 +106,18 @@ extern int unittests_fatal;
         return 1; \
     } while (0)
 
+/**
+ * \brief Skip the test.
+ *
+ * Used to skip the tests that cannot be run in the current environment.
+ * The aim is to keep this at 0.
+ */
+#define SKIP(reason)                                                                               \
+    do {                                                                                           \
+        SCLogInfo("Test skipped: %s", reason);                                                     \
```

Comment:
no need to `SCLogInfo` as we will have the `printf("skip\n");`  after ?

Reply:
```
./src/suricata -u -U DPDKRunmodeSetThreads01
Test DPDKRunmodeSetThreads01                                      : Info: affinity: Test skipped: not enough cpus in the system, the test requires at least 32 cores [UnitTestsUtilAffinityVerifyCPURequirement:util-affinity.c:347]
skip
==== TEST RESULTS ====
PASSED: 0
SKIPPED: 1
FAILED: 0
======================
```

This was the idea - to only tell the user that the test was skipped but a responsible test writer should explain why the test was skipped and what prerequirements were not met for the test.

Reply:
The macro SKIP_INCOMPATIBLE_ENVIRONMENT could skip the tests not only based on CPU count but potentially on other requirements as well.